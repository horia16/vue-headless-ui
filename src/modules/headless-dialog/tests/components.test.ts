import { describe, expect, it, vi } from "vitest";
import { mount } from "@vue/test-utils";
import { HDialog, HDialogContent, HDialogTitle, HDialogDescription } from "../";

const dialog = defineComponent({
  components: {
    HDialog,
    HDialogContent,
    HDialogTitle,
    HDialogDescription
  },
  props: {
    openByDefault: { type: Boolean, default: false }
  },
  setup(props) {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const open = ref(props.openByDefault);

    return { open };
  },
  template: /*html*/ `
  <div>
    <button @click="open = true">Open</button>
  </div>
  <h-dialog id='fixed_id' v-model="open">
    <h-dialog-content>
      <button @click="open = false">close_button</button>
      <h-dialog-title html-tag="h1">dialog_title</h-dialog-title>
      <h-dialog-description html-tag="p">dialog_descriotion</h-dialog-description>
    </h-dialog-content>
  </h-dialog>
  `
});

describe("dialog", () => {
  it("should open", async () => {
    const wrapper = mount(dialog, { attachTo: "body" });
    const button = wrapper.find("button");
    await button.trigger("click");
    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it("should close", async () => {
    const wrapper = mount(dialog, { props: { openByDefault: true }, attachTo: "body" });
    const button = wrapper.find("button");
    await button.trigger("click");
    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it("should trap the focus", async () => {
    //
  });

  it("should have aria description and title", async () => {
    //
  });
});
