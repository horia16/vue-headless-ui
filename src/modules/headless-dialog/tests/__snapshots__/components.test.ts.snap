// Vitest Snapshot v1

exports[`dialog > should close 1`] = `"<div data-v-app=\\"\\"><div><button>Open</button></div><!--teleport start--><!--teleport end--></div><transition-stub><div id=\\"fixed_id\\" style=\\"position: fixed; width: 100vw; height: 768px; z-index: 100;\\" class=\\"\\" aria-modal=\\"true\\" aria-labelledby=\\"fixed_id_title\\" aria-describedby=\\"fixed_id_description\\"><div role=\\"document\\"><button>close_button</button><h1 id=\\"fixed_id_title\\">dialog_title</h1><p id=\\"fixed_id_description\\">dialog_descriotion</p></div></div></transition-stub><div data-v-app=\\"\\"><div><button>Open</button></div><!--teleport start--><!--teleport end--></div><transition-stub><div id=\\"fixed_id\\" style=\\"position: fixed; width: 100vw; height: 768px; z-index: 100;\\" class=\\"\\" aria-modal=\\"true\\" aria-labelledby=\\"fixed_id_title\\" aria-describedby=\\"fixed_id_description\\"><div role=\\"document\\"><button>close_button</button><h1 id=\\"fixed_id_title\\">dialog_title</h1><p id=\\"fixed_id_description\\">dialog_descriotion</p></div></div></transition-stub>"`;

exports[`dialog > should open 1`] = `"<div data-v-app=\\"\\"><div><button>Open</button></div><!--teleport start--><!--teleport end--></div><transition-stub><div id=\\"fixed_id\\" style=\\"position: fixed; width: 100vw; height: 768px; z-index: 100;\\" class=\\"\\" aria-modal=\\"true\\" aria-labelledby=\\"fixed_id_title\\" aria-describedby=\\"fixed_id_description\\"><div role=\\"document\\"><button>close_button</button><h1 id=\\"fixed_id_title\\">dialog_title</h1><p id=\\"fixed_id_description\\">dialog_descriotion</p></div></div></transition-stub>"`;
